Rspec tests

rsundar opened the pull request:
Please review the rspec tests for my Tic Tac Toe implementation.

I am working alone since there is no one available in the same time zone as I am for this course.

Please don't hesitate to contact me on slack ( @rsundar ) if any significant changes are required.

Thanks

rsundar committed:
First commit for the rspec tests.

rsundar committed:
spec_helper.rb created.

rsundar committed:
Added Player tests, tests passed

rsundar committed:
Changed endings from CRLF to LF

rsundar committed:
added tests to check if a default name is assigned to the player object.

rsundar committed:
rspec tests for the board.

rsundar committed:
Finished board tests, all tests passing.

rsundar committed:
Minor formatting of tests for board_spec

rsundar committed:
Added rspec tests for the Game class.

wathigo commented on spec/board_spec.rb:
```diff
@@ -0,0 +1,107 @@
+require 'rspec'
+require '../lib/board.rb'
```
**[OPTIONAL]**
* Let's use `require_relative` instead of `require` in order to run the tests from the root directory.

rsundar replied:
Done!

wathigo reviewed (changes requested):
# Changes Required 
## Great work so far :clap:  
### [REQUIRED]
* Let's make sure that all the public methods in player, game, and board classes has been tested. See this :point_down: 
![uaw](https://user-images.githubusercontent.com/44072711/69747328-2c642480-1114-11ea-8f52-7d38d3f8f5a9.png).
Use this template to guide you :point_down: 
```
describe Board do
  describe '#update_board' do
    it '....' do
      # some code here
    end
    it '....' do
      # some code here
    end
  end
end
```
### After implementing the changes, submit another request for review.
**NB:** _All the current tests are okay, let's add some more to ensure all the public methods have been tested. Feel free to reach out on slack for any issue or clarification_

**Happy Coding :computer: 
Simon Wathgo**

rsundar committed:
changed require in board_spec to require_relative

rsundar committed:
refactored payer_spec to more accurately group similar tests.

rsundar committed:
changed all context to describe, and added tests for the is_valid? method in game_spec.

rsundar commented:
# Summary of Changes Made

As per the recommendations made by the previous reviewer ( @wathigo ) the following changes have been made:

- Changed the rspec tests in all of the spec files to reflect the recommended template of
  ```
      describe "#method" do
               it "..." do
                    #code here
               end
      end 
``` ```
- Added tests for the ```is_valid?``` method in the ```Game``` class.

Please contact me on Slack if any further changes are required. ( @rsundar )

maelfosso commented on spec/board_spec.rb:
```diff
@@ -0,0 +1,107 @@
+require 'rspec'
```
Change it with 
```ruby
require 'spec_helper'
```

rsundar replied:
Done

maelfosso commented on spec/game_spec.rb:
```diff
@@ -0,0 +1,100 @@
+require 'rspec'
```
Change it with 
```ruby
require 'spec_helper'
```

rsundar replied:
done

maelfosso commented on spec/player_spec.rb:
```diff
@@ -0,0 +1,36 @@
+require 'rspec'
```
Change it with 
```ruby
require 'spec_helper'
```

rsundar replied:
done

maelfosso reviewed (changes requested):
# Change required

Hello @rsundar 

Thanks for the changes, the improvements are awesome.
However, I had a problem when running the test. 
```ruby
$ rspec

An error occurred while loading ./spec/board_spec.rb. - Did you mean?
                    rspec ./spec/game_spec.rb
                    rspec ./spec/player_spec.rb

Failure/Error: require 'rspec'

LoadError:
  cannot load such file -- rspec
# ./spec/board_spec.rb:1:in `<top (required)>'

An error occurred while loading ./spec/game_spec.rb. - Did you mean?
                    rspec ./spec/board_spec.rb
                    rspec ./spec/player_spec.rb

Failure/Error: require 'rspec'

LoadError:
  cannot load such file -- rspec
# ./spec/game_spec.rb:1:in `<top (required)>'

An error occurred while loading ./spec/player_spec.rb. - Did you mean?
                    rspec ./spec/board_spec.rb
                    rspec ./spec/game_spec.rb

Failure/Error: require 'rspec'

LoadError:
  cannot load such file -- rspec
# ./spec/player_spec.rb:1:in `<top (required)>'
No examples found.


Finished in 0.00003 seconds (files took 0.33212 seconds to load)
0 examples, 0 failures, 3 errors occurred outside of examples
```
Please use **spec_helper** file. I suggest the changes to do to solve it

Kindly make that changes and submit a new code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

rsundar committed:
changed from require 'rspec' to require 'spec_helper'

rsundar commented:
# Summary of Changes Made

Corrected the error pointed out by @maelfosso 

- Changed all of the ```require 'rspec'``` at the beginning of each of the ```board_spec```,```game_spec```, and ```player_spec``` files in the ```spec``` directory to ```require 'spec_helper'```.

whiz25 reviewed (changes requested):
## Good job :+1: 
You did well implementing the suggested changes. I tested your code and am impressed most cases have passed. However, we need to fix a few edge cases before we can merge this one.
Please check my comments and make the requested changes.

### Martin Afani

whiz25 commented on spec/board_spec.rb:
```diff
@@ -0,0 +1,107 @@
+            expect(board.cells).to eq(winner)
+        end
+
+        it "Check if the board has a winner. Should return true" do
```
- [x] Board Winning Board Check if the board has a winner. Should return true

rsundar replied:
it does return true, that is the essence of the test.

Please clarify.

whiz25 commented on spec/board_spec.rb:
```diff
@@ -0,0 +1,107 @@
+            expect(board.winner?).to be true
+        end
+
+        it "The draw? method should return false" do
```
- [x] Board Winning Board The draw? method should return false

whiz25 commented on spec/board_spec.rb:
```diff
@@ -0,0 +1,107 @@
+            expect(board.moves).to eq(0)
+        end
+
+        it "Check if the game is a draw. Should be true" do
```
- [x] Board Drawn game Check if the game is a draw. Should be true

whiz25 commented on spec/board_spec.rb:
```diff
@@ -0,0 +1,107 @@
+            expect(board.draw?).to be true
+        end
+
+        it "The winner? method should return false" do
```
- [x] Board Drawn game The winner? method should return false

whiz25 commented on spec/game_spec.rb:
```diff
@@ -0,0 +1,100 @@
+            expect(game.player2.marker).to eq(marker2)
+        end
+
+        it "If new game is initialized the game state should be :active" do
```
- [x] Game Retrieve player names and markers If new game is initialized the game state should be :active

whiz25 commented on spec/game_spec.rb:
```diff
@@ -0,0 +1,100 @@
+        let(:board) { Board.new(win,9) }
+        subject(:game) { Game.new(player1, player2, current_player, board) }
+
+        it "Check if there is a winner. game_state should return :winner" do
```
- [x] Game Game Won Check if there is a winner. game_state should return :winner

whiz25 commented on spec/game_spec.rb:
```diff
@@ -0,0 +1,100 @@
+        let(:board) { Board.new(drawn,0) }
+        subject(:game) { Game.new(player1, player2, current_player, board) }
+
+        it "Check if the game is a draw. game_state should return :draw" do
```
- [x] Game Game Drawn Check if the game is a draw. game_state should return :draw

whiz25 reviewed (approved):
## Hey @rsundar 
Thanks for reaching out to me on slack. I was able to rectify the issue on my PC and now all tests pass as expected. So it's a :+1: from me. Merge and proceed to the next project.

### Martin Afani

rsundar commented:
Thanks for all of your help and patience @maelfosso @whiz25 @wathigo. It's greatly appreciated :smile:
